microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/MetricsTest.java

```diff
@@ -0,0 +1,320 @@
+
+import java.net.URL;
+import java.util.Arrays;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.eap.qe.microprofile.metrics.hello.MetricsApp;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+
+@RunWith(Arquillian.class)
+public class MetricsTest {
+
+    @ArquillianResource
+    URL deploymentUrl;
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    private RequestSpecification jsonMetricsRequest;
+    private RequestSpecification textMetricsRequest;
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class, MetricsTest.class.getSimpleName() + ".war")
+                .addPackage(MetricsApp.class.getPackage())
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Before
+    public void prepare() {
+        jsonMetricsRequest = given()
+                .baseUri("http://" + managementClient.getMgmtAddress())
+                .port(managementClient.getMgmtPort())
+                .basePath("metrics")
+                .accept(ContentType.JSON);
+        textMetricsRequest = given()
+                .baseUri("http://" + managementClient.getMgmtAddress())
+                .port(managementClient.getMgmtPort())
+                .basePath("/metrics")
+                .accept(ContentType.TEXT);
+    }
+
+    /**
+     * @tpTestDetails Smoke scenario to verify correct behaviour
+     * @tpPassCrit metrics are registered
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    public void applicationMetricsAreRegisteredAtDeploymentTime() {
+        jsonMetricsRequest.options().then()
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("$", hasKey("application"),
+                        "application", hasKey("hello-count"),
+                        "application", hasKey("hello-time"),
+                        "application", hasKey("hello-freq"),
+                        "application", hasKey("hello-invocations"),
+                        "application.keySet()", hasSize(4));
+    }
+
+    /**
+     * @tpTestDetails Smoke scenario to verify correct behaviour
+     * @tpPassCrit Correct metadata are provided for metrics
```

Comment:
What is the condition to decide if metadata are correct?

Reply:
Metadata are correct against the specification.